aws-flow/lib/aws/decider/data_converter.rb

```diff
@@ -26,7 +26,7 @@ class YAMLDataConverter
       #
       def dump(object)
         if object.is_a? Exception
-          return YAML.dump_stream(object, object.backtrace)
```

Comment:
When we dump_stream those objects it create an alias for the second object (object.backtrace) pointing to the backtrace property of the first object, this alias fails to load later on the load method. See (*1 -> &1)
```
2.7.3 :018 > YAML.dump_stream(exemplar, exemplar.backtrace)
 => "--- !ruby/exception:Exception\nmessage: exception\nbacktrace: &1\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/workspace.rb:114:in `eval'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/workspace.rb:114:in `evaluate'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/context.rb:459:in `evaluate'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:541:in `block (2 levels)\n  in eval_input'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:704:in `signal_status'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:538:in `block in eval_input'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/ruby-lex.rb:166:in `block\n  (2 levels) in each_top_level_statement'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/ruby-lex.rb:151:in `loop'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/ruby-lex.rb:151:in `block\n  in each_top_level_statement'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/ruby-lex.rb:150:in `catch'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb/ruby-lex.rb:150:in `each_top_level_statement'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:537:in `eval_input'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:472:in `block in run'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:471:in `catch'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:471:in `run'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/2.7.0/irb.rb:400:in `start'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in\n  `<top (required)>'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/bin/irb:23:in `load'\"\n- \"/Users/toshio/.rvm/rubies/ruby-2.7.3/bin/irb:23:in `<main>'\"\n--- *1\n"
```